SupportSDKSamples/CocoaPodSample/ViewController.swift

```diff
@@ -0,0 +1,169 @@
+    @IBOutlet weak var addImageButton: UIButton!
+    @IBOutlet weak var customFieldTextField: UITextField!
+    
+    var hcConfig: HelpCenterUiConfiguration {
+        let hcConfig = HelpCenterUiConfiguration()
+        hcConfig.hideContactSupport = true
+        hcConfig.labels = ["Breakout"]
+        return hcConfig
+    }
+    
+    var requestConfig: RequestUiConfiguration {
+        let config = RequestUiConfiguration()
+        config.tags = ["Some","stupid", "tags"]
+        config.subject = "Support_API Sample"
+        return config
+    }
+
+    @IBAction func openHC(_ sender: Any) {
+        let helpCenter = HelpCenterUi.buildHelpCenterOverview(withConfigs: [hcConfig, requestConfig])
+        navigationController?.pushViewController(helpCenter, animated: true)
+    }
+    
+    @IBAction func openRequestList(_ sender: Any) {
+        let requestList = RequestUi.buildRequestList(with: [requestConfig])
+        self.navigationController?.pushViewController(requestList, animated: true)
+    }
+
+    // MARK: Request Creation
+    
+    func buildCreateRequest(callback: @escaping (ZDKCreateRequest) -> Void) {
+        let request = ZDKCreateRequest()
+        request.subject = subjectTextField.text ?? requestConfig.subject
+        request.requestDescription = descriptionTextView.text ?? "Default: The problem of the ticket"
+        request.tags = requestConfig.tags
+        request.customTicketFields = getCustomFields()
+        
+        uploadAttachment { (attachment) in
+            if let attachment = attachment {
+                request.attachments.append(attachment)
+            }
+            callback(request)
+        }
+    }
+    
+    private func getCustomFields() -> [ZDKCustomField] {
+        if let customFieldValue = customFieldTextField.text, !customFieldValue.isEmpty {
+            return [ZDKCustomField(fieldId: NSNumber(value: 1), andValue: customFieldValue)] // get fieldId from Support Settings
+        }
+        return []
+    }
+    
+    func uploadAttachment(callback: @escaping (ZDKUploadResponse?) -> Void) {
+        ZDKUploadProvider().uploadAttachment(getImageDataFrom(view: attachmentImageView), withFilename: "image_name_app.png", andContentType: "image")
+        { (response, error) in
+            if let response = response {
+                print("Token: \(response.uploadToken.debugDescription)")
+                print("Attachment: \(response.attachment.debugDescription)")
+            }
+            if let error = error {
+                print("Error: \(error.localizedDescription)")
+            }
+            callback(response)  // response always gets sent regardless if nil.
+        }
+    }
+    
+    @IBAction func sendRequest(_ sender: Any) {
+        buildCreateRequest { (request) in
+            ZDKRequestProvider().createRequest(request)  { (result, error) in
+                if result != nil {
+                    self.showToast(with: "Message Sent")
```

Comment:
This closure will cause `self` to leak. 
Use `[unowned self]` or `[weak self]` notation to avoid memory leaks. Within Swift 4.2 you can define strong self with this notation
```guad let self = self else { return } ```